src/transform/plugins/images/index.ts

```diff
@@ -124,10 +174,47 @@ const index: MarkdownItPluginCb<Opts> = (md, opts) => {
 
                     childrenTokens[j].attrSet('yfm_patched', '1');
                 }
+                j++;
+            }
+
+            j = 0;
+            const newTokens: Token[] = [];
 
+            while (j < childrenTokens.length) {
```

Comment:
Replace default token `image` with `image_with_caption` can be wery destructive for external plugins, which searches for `image` token only.

I propose to implement this in other way:

- Check what image token has `caption` attr.
- Then generate new token `image_caption` after this one.
- Add renderer for `image_caption` which will generate all required a11y compatible html.
- Convert `caption` attr on image to some class attr `image.attrSet('class', image.attrGet('class') + ' yfm-image-with-caption')`
- Add some styles to main css.

Reply:
Makes sense, I got rid of `image_with_caption` token

Reply:
Wdyt, does the current solution work or potentially could have a negative impact on other image-related plugins?
What about the CSS? Do I need to align it to the left?
<img width="271" alt="Screenshot 2024-12-18 at 19 01 38" src="https://github.com/user-attachments/assets/6f32a151-50b5-4ea5-b38f-6afc13bdcfef" />